src/Middleware/InjectDetector.php

```diff
@@ -28,12 +28,11 @@ public function handle(Request $request, \Closure $next)
             // no-op.
         }
 
-        if (! $response instanceof Response) {
-            return $next($request);
+        // Inject the detector when returning an HTTP response
+        if ($response instanceof Response) {
+            $this->detector->injectDetector($request, $response);
         }
 
-        $this->detector->injectDetector($request, $response);
-
         return $response;
```

Comment:
I'm not sure if you can just leave out the `$next()` because other middleware need to be called as well.

Reply:
My understanding is that at this point `$next()` was already called on line 26; calling it on line 32 will duplicate the execution of the remainder of the request.

Reply:
I suppose it might make more sense to make this decision based on whether the request is expecting JSON - I will update the PR.